TypeInference.md

```diff
@@ -85,3 +85,27 @@ class CustomLabel {
     let textColor = UIColor.white
 }
 ```
+
+## Casting
```

Comment:
Great to have more conventions on casting! There's a separate page on type casting: TypeCasting.md. Could we add more information to that page?

Reply:
Ah, my bad. I totally missed that page. I'll move this there.

Reply:
Actually, looks like you've already pulled in all the relevant information. I'm gonna close this one.